httpx/client.py

```diff
@@ -145,6 +145,8 @@ def merge_headers(
         if auth is not None:
             if isinstance(auth, tuple):
                 auth = HTTPBasicAuth(username=auth[0], password=auth[1])
+            elif not isinstance(auth, BasicAuthBase):
```

Comment:
This check isn't correct here - any `AuthBase` subclass is valid, not just `BasicAuthBase`